### ISSUE TITLE
FHIR $health-cards-issue operation fails 

### ISSUE THREAD

**Issue by gjskha** [dvci/health_cards#95]
Possibly related to Issue #94 . I created a test script as follows:

```
cat <<EOF >"/tmp/data"
{
  "resourceType": "Parameters",
  "parameter": [{
    "name": "credentialType",
    "valueUri": "https://smarthealth.cards#covid19"
  }]
}
EOF

curl -X POST 'https://<myinstance>/Patient/8/$health-cards-issue' \
-H 'Content-Type: application/fhir+json' -H 'Accept: application/fhir+json' \
-d @/tmp/data
```
I get back the following response:

```
{
  "issue": [
    {
      "severity": "error",
      "code": "exception"
    }
  ],
  "resourceType": "OperationOutcome"
}
```
The logs don't have much detail in this case.

```
Processing by HealthCardsController#create as FHIR_JSON
Parameters: {"patient_id"=>"8"}
Completed 500 Internal Server Error in 21ms (Views: 0.1ms | ActiveRecord: 7.6ms | Allocations: 8348)
```

**Comment by ssayer** [comment]
As with #94, please make sure you're on the latest version. If that doesn't fix the issue, please look in `app/controllers/application_controller.rb` and comment out all lines in `handle_fhir_errors` except` for the `yield`. This should suppress the error handling and give you the full stack trace. I suspect it is the same issue, but this will help to confirm.

**Comment by ssayer** [comment]
Closing as with #94

**Comment by gjskha** [comment]
OK. I pulled the latest changes and commented out the specified lines in `handle_fhir_errors`. What I have discovered is that session[:patient_id] value is missing inside of the `health_cards` method of the application_controller file when fetching the `show` controller for Patient, but only with calls subsequent to the patient's creation. This causes @health_card to be empty, which causes the symptom I described. 

I have not made any significant code changes to my sandbox, so I am curious if this can be replicated. If so, then presumably the real fix is to create more robust session management, but maybe there is a quick fix just for the sake of the reference implementation.
